src/main/java/com/atlassian/db/replica/internal/ConsistencyToTransactionHookAdapter.java

```diff
@@ -0,0 +1,24 @@
+package com.atlassian.db.replica.internal;
+
+import com.atlassian.db.replica.spi.ReplicaConsistency;
+import com.atlassian.db.replica.spi.TransactionHook;
+
+import java.sql.Connection;
+
+public class ConsistencyToTransactionHookAdapter implements TransactionHook {
```

Comment:
So `ReplicaConsistency` doesn't implement `TransactionHook`, because we want to make `preCommit` explicit rather than implicitly no-op?

Reply:
Yes. No-op was there not to break the API.